opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/SpanShim.java

```diff
@@ -214,10 +238,40 @@ static Attributes convertToAttributes(Map<String, ?> fields) {
       } else if (value instanceof Boolean) {
         attributesBuilder.put(booleanKey(key), (Boolean) value);
       } else {
-        attributesBuilder.put(stringKey(key), value.toString());
+        AttributeKey<String> attributeKey = null;
+        if (isError && !isRecordingException) {
+          if (key.equals(Fields.ERROR_KIND)) {
+            attributeKey = SemanticAttributes.EXCEPTION_TYPE;
+          } else if (key.equals(Fields.MESSAGE)) {
+            attributeKey = SemanticAttributes.EXCEPTION_MESSAGE;
+          } else if (key.equals(Fields.STACK)) {
+            attributeKey = SemanticAttributes.EXCEPTION_STACKTRACE;
+          }
+        }
+        if (!isRecordingException
+            || !key.equals(Fields.ERROR_OBJECT)
+            || !(value instanceof Throwable)) {
+          if (attributeKey == null) {
+            attributeKey = stringKey(key);
+          }
+          attributesBuilder.put(attributeKey, value.toString());
+        }
       }
     }
 
     return attributesBuilder.build();
   }
+
+  @Nullable
+  private static Throwable findThrowable(Map<String, ?> fields) {
+    for (Map.Entry<String, ?> entry : fields.entrySet()) {
```

Comment:
Instead of looping can just be `fields.get(Fields.ERROR_OBJECT)` right?

Reply:
done